### PR TITLE
feat(item-popup): Adding Catalyst Progress for unlocked, incomplete, catalysts on the item popup (8771)

### PR DESCRIPTION
**Pull request by mattjbones**
**Problem** 
The item popup doesn't show any progress information for unlocked, but incomplete, exotic weapon catalysts as raised in issue #8771. 

**Solution**
Following the existing `WeaponDeepsightInfo`  component for inspiration a new section can be added to the item popup. 
Which can be populated by pulling out the necessary objectives from `buildCatalystInfo` function.
The progress only displaying if the user has the unlocked catalyst with available objectives (otherwise there's nothing shown).

Screenshot 2022-10-20 at 21 06 14

Screenshot 2022-10-20 at 21 06 06

**note**
There were additional changes that came along with the `yarn i18n` command, "undo" and "redo" - not really sure what the best action here is?

### CODE REVIEW

File: src/app/inventory/item-types.ts [hunk 1]
```diff
@@ -282,9 +282,10 @@ export interface DimCatalyst {
   complete: boolean;
   /** Whether the player has unlocked/discovered the catalyst */
   unlocked: boolean;
+  /** Progress the player has made on unlocking the catalyst  */
+  objectives?: DestinyObjectiveProgress[];
 
   // TODO: on item popup, remove catalyst socket if item.catalystInfo is undefined
```
Comment (delphiactual) [inline src/app/inventory/item-types.ts:288]: this todo can be removed as well. the catalyst socket does not show on all items anymore

File: src/locale/en.json [hunk 2]
```diff
@@ -639,6 +639,7 @@
     "RandomizeSearch": "Randomize from Search",
     "RandomizeSearchPrompt": "Randomize your equipped items from search \"{{query}}\"?",
     "RandomizeWeapons": "Randomize your equipped weapons?",
+    "Redo": "",
```
Comment (delphiactual) [inline src/locale/en.json:642]: @bhollis looks like you forgot to add your i18n strings for undo/redo

**Comment by delphiactual** [comment]
not seeing any test failures on a clean master, install and test

**Review by delphiactual — APPROVED** [review]

**Comment by mattjbones** [comment]
> not seeing any test failures on a clean master, install and test
> 
> EDIT: Tests are all passing on this PR too https://github.com/DestinyItemManager/DIM/pull/8792/checks?check_run_id=9016891706

Ok, that could be an env issue - failing on this line `return JSON.parse(await fs.readFile(filename, 'utf-8'));` due to "Unexpected end of JSON input", looks like it's trying to load some HUUUUGE JSON and for whatever reason it was borked on my machine. Anyway I deleted it (to force it to redownload) and everything appears to be passing locally now.

**Review by bhollis — APPROVED** [review]
Awesome, thanks for doing this!

**Pull request merged**